libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/WayNameView.java

```diff
@@ -41,9 +49,10 @@ public String retrieveWayNameText() {
   }
 
   public void updateVisibility(boolean isVisible) {
-    int visibility = isVisible ? VISIBLE : INVISIBLE;
-    if (getVisibility() != visibility) {
-      setVisibility(visibility);
+    if (isVisible && !TextUtils.isEmpty(wayNameText.getText())) {
```

Comment:
I have concern here lets assume following steps:
- `updateWayNameText(String.EMPTY_STRING)`
- `updateVisibility(true)` 
- `updateWayNameText("OLOLO")`

In this case view is not showed up. Does it expected behaviour?

Reply:
the same for `WayNameView.java` at new ui-module

Reply:
This PR's first version had the logic that to "show or hide" the view when setting the text. 

We could argue that's not good too. It is against the "single responsibility principle" that the _updateText_ method also updates the visibility. Consumer will confuse. 
Another scenario is when user switches to "overall view mode" from "center" mode, we will hide this wayNameView by calling `updateVisitbility(false)`. If user doesn't remove the `OnWayNameChangedListener` then it will keep updating the text which later will show this view in the "overall view mode".

And third is, do you think `updateWayNameText(String.EMPTY_STRING)` is a valid call by developer? What's the meaning to do so? Just try to hide the wayNameView? I would assume they should follow steps to do it:
* `updateWayNameText(String.EMPTY_STRING)`
* `updateVisibility(false)`

or just call `updateVisibility(false)`

@RingerJK , wdyt?

Reply:
My point, this is unexpected behaviour in my comment above. 
I like your approach when customer have to call hide/show view explicitly through `updateVisibility ` 👍